src/config.ts

```diff
@@ -0,0 +1,171 @@
+     * @since 1.7.14
+     * @type Number
+     * @default
+     */
+    maxCacheSideLimit = 4096
+
+    /**
+     * Lowest pixel limit for cache canvases, set at 256PX
+     * @since 1.7.14
+     * @type Number
+     * @default
+     */
+    minCacheSideLimit = 256
+
+    /**
+     * When 'true', style information is not retained when copy/pasting text, making
+     * pasted text use destination style.
+     * Defaults to 'false'.
+     * @type Boolean
+     * @default
+     * @deprecated
+     */
+    disableStyleCopyPaste = false
+
+    /**
+     * Enable webgl for filtering picture is available
+     * A filtering backend will be initialized, this will both take memory and
+     * time since a default 2048x2048 canvas will be created for the gl context
+     * @since 2.0.0
+     * @type Boolean
+     * @default
+     */
+    enableGLFiltering = true
+
+    /**
+     * if webgl is enabled and available, textureSize will determine the size
+     * of the canvas backend
+     * 
+     * In order to support old hardware set to `2048` to avoid OOM
+     * 
+     * @since 2.0.0
+     * @type Number
+     * @default
+     */
+    textureSize = 4096
+
+    /**
+     * Skip performance testing of setupGLContext and force the use of putImageData that seems to be the one that works best on
+     * Chrome + old hardware. if your users are experiencing empty images after filtering you may try to force this to true
+     * this has to be set before instantiating the filtering backend ( before filtering the first image )
+     * @type Boolean
+     * @default false
+     */
+    forceGLPutImageData = false
+
+    /**
+     * WebGL
+     */
+    maxTextureSize?: number
+
+    /**
+     * WebGL
+     */
+    webGLPrecision?: TWebGLPrecision
+
+    /**
+     * If disabled boundsOfCurveCache is not used. For apps that make heavy usage of pencil drawing probably disabling it is better
+     * @default true
+     */
+    cachesBoundsOfCurve = true
+
+    /**
+     * Map of font files
+     * Map<fontFamily, pathToFile> of font files
+     */
+    fontPaths: Record</** fontFamily */ string, /** pathToFile */ string> = {}
+
+    /**
+     * Defines the number of fraction digits to use when serializing object values.
+     * Used in exporting methods (`toObject`, `toJSON`, `toSVG`)
+     * You can use it to increase/decrease precision of such values like left, top, scaleX, scaleY, etc.
+     */
+    NUM_FRACTION_DIGITS = 4
+}
+
+export class Configuration extends BaseConfiguration {
+
+    constructor(config?: TConfiguration) {
+        super();
+        this.configure(config);
+    }
+
+    configure(config: TConfiguration = {}) {
+        Object.assign(this, config);
+    }
+
+    /**
+     * Map<fontFamily, pathToFile> of font files
+     */
+    addFonts(paths: Record</** fontFamily */ string, /** pathToFile */ string> = {}) {
+        this.fontPaths = {
+            ...this.fontPaths,
+            ...paths
+        };
```

Comment:
Object.assign(this.fontPaths, paths) in general seems nicer to me, but i don't care.

Reply:
This is really something that should be local IMO

Reply:
Only used in canvas.toSVG